swap/protocol_test.go

```diff
@@ -16,105 +16,236 @@
+			{
+				Code: 0,
+				Msg: &HandshakeMsg{
+					ContractAddress: swap.owner.Contract,
+				},
+				Peer: creditor.ID(),
+			},
+			{
+				Code: 0,
+				Msg: &HandshakeMsg{
+					ContractAddress: swap.owner.Contract,
+				},
+				Peer: debitor.ID(),
+			},
+		},
+	})
+
+	// there should be no error at this point
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+// TestEmitCheque is a full round of a cheque exchange between peers via the protocol.
+// We create two swap, for the creditor (beneficiary) and debitor (issuer) each,
+// and deploy them to the simulated backend.
+// We then create Swap protocol peers with a MsgPipe to be able to directly write messages to each other.
+// We have the debitor send a cheque via an `EmitChequeMsg`, then the creditor "reads" (pipe) the message
+// and handles the cheque. We finally also send back a `ConfirmMsg` to the debitor
+// TODO: The ConfirmMsg part is not definitevely specificied.
+func TestEmitCheque(t *testing.T) {
+	log.Debug("set up test swaps")
+	creditorSwap, testDir1 := newTestSwap(t)
+	debitorSwap, testDir2 := newTestSwap(t)
+	defer os.RemoveAll(testDir1)
+	defer os.RemoveAll(testDir2)
+
+	ctx := context.Background()
+
+	log.Debug("deploy to simulated backend")
+	testDeploy(ctx, creditorSwap.backend, creditorSwap)
+	testDeploy(ctx, debitorSwap.backend, debitorSwap)
+	creditorSwap.backend.(*backends.SimulatedBackend).Commit()
+	debitorSwap.backend.(*backends.SimulatedBackend).Commit()
+
+	log.Debug("create peer instances")
+	// create Peer instances
+	// NOTE: remember that these are peer instances representing each **a model of the remote peer** for every local node
+	// so creditor is the model of the remote mode for the debitor! (and vice versa)
+
+	// in order to be able to model as realistically as possible sending and receiving, let's use a MsgPipe
+	// a MsgPipe is a duplex read-write object, write to one end and read from the other
+
+	// create the message pipe
+	crw, drw := p2p.MsgPipe()
+	// create the creditor peer
+	cPtpPeer := p2p.NewPeer(enode.ID{}, "creditor", []p2p.Cap{})
+	cProtoPeer := protocols.NewPeer(cPtpPeer, crw, Spec)
+	// create the debitor peer
+	dPtpPeer := p2p.NewPeer(enode.ID{}, "dreditor", []p2p.Cap{})
+	dProtoPeer := protocols.NewPeer(dPtpPeer, drw, Spec)
+	// create the Swap protocol peers
+	creditor := NewPeer(cProtoPeer, debitorSwap, debitorSwap.backend, creditorSwap.owner.address, debitorSwap.owner.Contract)
+	debitor := NewPeer(dProtoPeer, creditorSwap, creditorSwap.backend, debitorSwap.owner.address, debitorSwap.owner.Contract)
+
+	// a safe check: at this point no cheques should be in the swap
+	if len(creditorSwap.cheques) != 0 {
+		t.Fatalf("Expected no cheques at creditor, but there are %d:", len(creditorSwap.cheques))
+	}
+
+	log.Debug("create a cheque")
+	var err error
+	cheque := &Cheque{
+		ChequeParams: ChequeParams{
+			Contract:    debitorSwap.owner.Contract,
+			Beneficiary: creditorSwap.owner.address,
+			Amount:      42,
+			Timeout:     0,
```

Comment:
Since you added `Honey` as a new field to ChequeParams you should also add it here with value `42` so that it is a valid cheque according to the current oracle.